packages/program/src/functions/base-invocation-scope.ts

```diff
@@ -75,7 +75,7 @@ export class BaseInvocationScope<TReturn = any> {
    */
   protected get calls() {
     const script = getContractCallScript(this.functionInvocationScopes);
-    const provider = this.program.provider as Provider;
+    const provider = this.getProvider();
```

Comment:
here is missing the validation ??
```

    assert(provider, 'Wallet or Provider is required!');
``` 

if provider is really mandatory for this class, getProvider could centralize the assert validation and remove the possibility of any `getProvider()` usage returning empty

Reply:
@LuizAsFight I am not 100% sure why these `Provider` validations are in place, or if they are needed. 

The way I see it, `this.program.provider` will never be undefined because `program` is an instance of `AbstractProgram`, meaning that this will be either a `Script` or `Contract` instance. 

After taking a look at these class constructors I've noticed that it is possible to instantiate a `Contract` without an `Account` but never without a `Provider`. To instantiate a `Script` the `Account` is required, therefore, the `Provider` instance will be always present.

My understanding is; `this.program.account` can be undefined, but `this.program.provider` will never be. 

Not sure if these assumptions are correct.

Reply:
Agreed, this validation seems redundant and not required.

On top of that, we should **_not_** use `assert` anymore, as everything is currently based on `FuelError`.